### ISSUE TITLE
Team wars and other ideas

### ISSUE THREAD

**Issue by Lurmey** [LatvianModder/FTBUtilities#712]
Hiya, loving using this on my server. I had a couple ideas;

Enemy teams can wage war on one another. Setting a team as an enemy is like rivalling their empire in Stellaris and provides them with a casus belli on you while preventing them entry to your lands. In order for them to declare war on your team, they must set your team as their enemy too, then you both have the option to declare war.

While at war, both teams are permitted entry into the others lands and can build/destroy blocks. If a chunk is occupied by a member of the attacking team for a certain period of time without being contested by a member or an ally, that chunk is said to be "occupied" by the attacking team.  If a certain threshold of chunk occupation is passed, say 70%, then the defending team is disbanded and all their land is given to the attacking team. At any point during this, either team's owner can surrender, giving the attacking team full control of the chunks they had occupied while returning any occupied or claimed chunks in the attacker's lands back to them.

The attacking team may try to enforce their demands on the defending team by requesting that they surrender. This could be part of the mod or simply be said in chat.

If no chunks are occupied on either side, or the occupation on each side is the same (team 1 claims 3 chunks from team 2 and vice versa) then either can declare a white peace. The other team has to accept a white peace for it to take effect, whereas surrender is in effect immediately without the attacker's acceptance. If one team offers a white peace and the other ignores it, then the occupation shifts to one side or the other, the white peace offer is disregarded.

I think this would be a much cooler way of engaging in PVP with others in a fairer and legitimate way. It would also allow people to claim and fight for their land without it being griefing, as both sides have to interact in order for a war to be a possibility. Though, declaring a team as your enemy only has the benefit of closing your borders to them, which should ideally be a different option, but then what advantage would there be to declaring an enemy? In Stellaris, you gain political influence. Perhaps there should be another advantage to setting a team as your enemy.

Also, currently it appears that you can only set a specific player as an enemy. Does this set their whole team as enemies or just the one player? Whole teams should be set as enemies, not individuals, but perhaps this should be a config option.

One other thing that I think would be helpful... Change the colour of a player's name tag depending on their team relation to you. Fellow members in blue, allies in green and enemies in red.

**Comment by LatvianModder** [comment]
Very interesting idea!
Only players can be set as enemies, but whatever team they are in will also be considered enemy. That is to prevent them from switching team to avoid being on enemy list

**Comment by Lurmey** [comment]
Thanks! I'm glad you think so.

Ah, thanks for letting me know that. I wasn't quite sure how it worked. I'm not sure whether I think that the current system is better than setting whole teams as enemy independently or not. Can you still set a player as an enemy even if they're not in a team? If that's so, then the current system is definitely better.

**Comment by LatvianModder** [comment]
Yes, alllies are team based, but work for all members in that team, but enemies are player based, but also work for all members of their team

**Comment by Colton-Slayden** [comment]
Any updates on this?
